### PR TITLE
[Doc] Revamp ray core design patterns doc [8/n]: pass large arg by value

### PR DESCRIPTION
**Pull request by jjyao**
Signed-off-by: Jiajun Yao <jeromeyjj@gmail.com>

## Why are these changes needed?
Add a new anti-pattern of passing large arg by value.

## Related issue number
https://github.com/ray-project/ray/issues/27048

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-core/doc_code/anti_pattern_pass_large_arg_by_value.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+ray.init()
+
+@ray.remote
+def func(large_arg, i):
+    return len(large_arg) + i
+
+large_arg = np.zeros(1024 * 1024)
+
+# 10 copies of large_arg is stored in the object store
```
Comment (stephanie-wang) [inline doc/source/ray-core/doc_code/anti_pattern_pass_large_arg_by_value.py:13]:
```suggestion
# 10 copies of large_arg are stored in the object store.
```

File: doc/source/ray-core/doc_code/anti_pattern_pass_large_arg_by_value.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+outputs = ray.get([func.remote(large_arg, i) for i in range(10)])
+# __anti_pattern_end__
+
+# __better_approach_start__
+# 1 copy of large_arg is stored in the object store
```
Comment (stephanie-wang) [inline doc/source/ray-core/doc_code/anti_pattern_pass_large_arg_by_value.py:18]:
```suggestion
# 1 copy of large_arg is stored in the object store.
```

File: doc/source/ray-core/patterns/pass-large-arg-by-value.rst [hunk 3]
```diff
@@ -0,0 +1,29 @@
+Anti-pattern: Passing the same large argument by value repeatly harms performance
```
Comment (stephanie-wang) [inline doc/source/ray-core/patterns/pass-large-arg-by-value.rst:1]:
```suggestion
Anti-pattern: Passing the same large argument by value repeatedly harms performance
```

File: doc/source/ray-core/patterns/pass-large-arg-by-value.rst [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+=================================================================================
+
+**TLDR:** Avoid passing the same large argument by value to multiple tasks, passing by reference instead.
```
Comment (stephanie-wang) [inline doc/source/ray-core/patterns/pass-large-arg-by-value.rst:4]:
```suggestion
**TLDR:** Avoid passing the same large argument by value to multiple tasks, use ``ray.put()`` and pass by reference instead.
```

**Review by stephanie-wang — APPROVED** [review]

**Review by jianoaix — COMMENTED** [review]
LGTM, thanks for adding this

File: doc/source/ray-core/doc_code/anti_pattern_pass_large_arg_by_value.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+large_arg_ref = ray.put(large_arg)
+outputs = ray.get([func.remote(large_arg_ref, i) for i in range(10)])
```
[the remark this answers is not in the document]
Comment (jjyao) [reply doc/source/ray-core/doc_code/anti_pattern_pass_large_arg_by_value.py:20]: Passed by PRC inline since it's smaller than 100KB.
